core/save-xml.c

```diff
@@ -544,7 +543,7 @@ static void save_one_device(void *_f, const char *model, uint32_t deviceid,
 	put_format(b, "/>\n");
 }
 
-int save_dives(const char *filename)
+int save_dives_file(const char *filename)
```

Comment:
the whole point was that the code "above" in the stack could just call "save_dives" and not worry if this is a local file, or git, or whatever. Not sure why we'd want to move that information "up"

Reply:
Sure, I get that. But now FileLocation is C++ and the corresponding multiplexer has to move one layer up. In the desktop version, these are the methods `loadDives` and `saveDives`. Indeed, I'm not really happy with them and hopefully they can be simplified, by e.g. moving the whole checkcloudonline thing out.

If I understand correctly, in the mobile version there is only git access, so there is no need to ever call the xml version. I think one could even not compile it for mobile?

Also note that the picture is not as rosy as you paint it - if it wants to check for SHA, the mobile code does not simply call `read_dive()`, instead it calls `check_git_sha()` and then `git_load_dives()`. Clearly, the current interface is less than optimal.

Reply:
There is a LONG standing request to allow people to read/write XML files in the mobile app. The idea is to say "hey, I don't care how you get the file here, dropbox, Google Drive, whatever... just read/write this XML file". No one ever bothered to write a UI for it, but now that I understand QML a bit better (famous last words) I know that I could do that without too much pain. Assuming Santa brings me a few extra hours per day, that is.

Also, I don't understand why FileLocation being C++ forces things to move up in the stack. We want to be very careful not to lose the abstraction that for our UI code the filename alone is all it needs to know in order to let the lower level decide where/how things get stored.